fe1-web/model/objects/WalletCryptographyHandler.ts

```diff
@@ -0,0 +1,76 @@
+import { IndexedDBStore, WalletCryptoKey } from 'store/stores/IndexedDBStore';
+import STRINGS from 'res/strings';
+
+/**
+ * This class has the job of handling the cryptography functions of the wallet.
+ * It interacts with the IndexedDB database in order to store and retrieve the
+ * secret key. It also encrypts and decrypts the tokens with the retrieved key.
+ */
+export class WalletCryptographyHandler {
+  /* encryption/decryption algorithm (RSA) */
+  private readonly algorithm = {
+    name: 'RSA-OAEP',
+    modulusLength: 4096,
+    publicExponent: new Uint8Array([1, 0, 1]),
+    hash: 'SHA-256',
+  };
+
+  private readonly keyUsages: KeyUsage[] = ['encrypt', 'decrypt'];
+
+  /* the encryption/decryption key storage */
+  private keyDatabase: IndexedDBStore = new IndexedDBStore();
+
+  /**
+   * this method adds an RSA encryption/decryption key,
+   * interacts with IndexedDB browser database.
+   */
+  public async initWalletStorage(): Promise <void> {
+    const key: WalletCryptoKey = await this.generateRSAKey();
+    await this.keyDatabase.putKey(key);
+  }
+
+  /**
+   * encrypts the given ed25519 token with the RSA key stored in the indexedDB database
+   * @param token ed25519 key toUint8Array()
+   */
+  public async encrypt(token: Uint8Array): Promise<ArrayBuffer> {
+    const key = await this.getKeyFromDatabase(STRINGS.walletPublicKey);
```

Comment:
to get a key, you could either do `await get(this.publicKeyId);` (with `get` defined as in idb-keyval),
or here you're calling another method, who then calls on the other class.

It seems there's a ton of indirection which only makes the actual behavior harder to analyze.